packages/lodestar-validator/src/services/attestation.ts

```diff
@@ -191,11 +198,16 @@ export class AttestationService {
     }
   }
 
-  private async waitForAttestationBlock(slot: Slot): Promise<void> {
-    this.logger.debug("Waiting for slot block", {slot});
+  private async waitForAttestationBlock(slot: Slot, signal: AbortSignal): Promise<void> {
+    this.logger.info("Waiting for slot block", {slot});
     const eventSource = new EventSource(`${this.provider.url}/lodestar/blocks/stream`, {
       https: {rejectUnauthorized: false},
     });
+
+    if (signal.aborted) {
+      this.logger.verbose("AttestationService: Abort waitForAttestationBlock");
+    }
```

Comment:
signal.on("abort", () => {clean(); resolve();}) should be inside promise, since with this, you cannot abort once it goes past this line

Reply:
what would clean() be in this case?  the clearTimeout?

Reply:
Yes